src/app/clients/add/formFunctions.ts

```diff
@@ -0,0 +1,281 @@
+    };
+};
+
+export const getBaby = (fieldSetter: FieldSetter, errorSetter: ErrorSetter): OnChange => {
+    return (event) => {
+        const input = event.target.value;
+        if (input === "Yes") {
+            errorSetter("nappySize", Error.initial);
+            fieldSetter("babyProducts", true);
+        } else {
+            errorSetter("nappySize", Error.none);
+            if (input === "No") {
+                fieldSetter("babyProducts", false);
+            } else {
+                fieldSetter("babyProducts", null);
+            }
+        }
+    };
+};
+
+export const errorExists = (errorType: Error): boolean => {
+    return errorType !== Error.initial && errorType !== Error.none;
+};
+
+export const errorText = (errorType: Error): string => {
+    return errorType === Error.initial ? Error.none : errorType;
+};
+
+export const checkboxGroupToArray = (checkedBoxes: booleanGroup): string[] => {
+    return Object.keys(checkedBoxes).filter((key) => checkedBoxes[key]);
+};
+
+export const formatPhoneNumber = (value: string): string => {
+    const numericInput = value.replace(/(\D)/g, "");
+    return numericInput[0] === "0" ? "+44" + numericInput.slice(1) : "+" + numericInput;
+};
+
+export const formatPostcode = (value: string): string => {
+    return value.replace(/(\s)/g, "").toUpperCase();
+};
+
+export const maxNumberChildren = (value: string): boolean => {
+    return parseInt(value) <= 20;
+};
+
+export const checkErrorOnSubmit = (
+    errorType: FormErrors,
+    errorSetter: React.Dispatch<React.SetStateAction<FormErrors>>
+): boolean => {
+    let errorExists = false;
+    let amendedErrorTypes = { ...errorType };
+    for (const [errorKey, error] of Object.entries(errorType)) {
+        if (error !== "none") {
+            errorExists = true;
+        }
+        if (error === "initial") {
+            amendedErrorTypes = {
+                ...amendedErrorTypes,
+                [errorKey]: "required",
+            };
+        }
+    }
+    if (errorExists) {
+        errorSetter({ ...amendedErrorTypes });
+    }
+    return errorExists;
+};
+
+export const insertFamily = async (
+    peopleArray: Person[],
+    familyID: string | null
+): Promise<boolean> => {
+    if (familyID === null) {
+        return false;
+    }
+    const familyRecords: FamilyDatabaseRecord[] = [];
+    for (const person of peopleArray) {
+        if (person.quantity === undefined || person.quantity > 0) {
+            familyRecords.push({
+                family_id: familyID,
+                person_type: person.personType,
+                quantity: person.quantity ?? 1,
+                age: person.age ?? null,
+            });
+        }
+    }
+    const { error: error } = await supabase.from("families").insert(familyRecords);
+    return error === null;
+};
+
+export const insertClient = async (clientRecord: ClientDatabaseRecord): Promise<string | null> => {
+    const { data: familyID, error: error } = await supabase
```

Comment:
`, error }` is sufficient (JS syntax for 'a key with string 'error' and value variable `error`)